Lib/test/test_json/test_speedups.py

```diff
@@ -36,6 +37,26 @@ def test_make_encoder(self):
             b"\xCD\x7D\x3D\x4E\x12\x4C\xF9\x79\xD7\x52\xBA\x82\xF2\x27\x4A\x7D\xA0\xCA\x75",
             None)
 
+    @test.support.cpython_only
+    def test_issue31505(self):
+        # There shouldn't be an assertion failure in case c_make_encoder()
+        # receives a bad encoder() argument.
+        def _bad_encoder1(*args):
+            return None
+        enc = self.json.encoder.c_make_encoder(None, None, _bad_encoder1, None,
```

Comment:
Use more meaningful arguments. Otherwise TypeError could be raised for different reason (for example if `default` is not a callable).
```
c_make_encoder(None, default, _bad_encoder1, None, ': ', ', ', False, False, False)
```